clouddriver-eureka/src/main/groovy/com/netflix/spinnaker/clouddriver/eureka/EurekaProviderConfiguration.groovy

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.eureka
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.clouddriver.eureka.api.EurekaApiFactory
+import com.netflix.spinnaker.clouddriver.eureka.provider.agent.EurekaAwareProvider
+import com.netflix.spinnaker.clouddriver.eureka.provider.agent.EurekaCachingAgent
+import com.netflix.spinnaker.clouddriver.eureka.provider.EurekaCachingProvider
+import com.netflix.spinnaker.clouddriver.eureka.provider.config.EurekaAccountConfigurationProperties
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.config.ConfigurableBeanFactory
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.ComponentScan
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.PropertySource
+import org.springframework.context.annotation.Scope
+
+import java.util.regex.Pattern
+
+@Configuration
+@EnableConfigurationProperties
+@ConditionalOnProperty('eureka.provider.enabled')
+@ComponentScan(["com.netflix.spinnaker.clouddriver.eureka"])
+@PropertySource(value = "classpath:META-INF/clouddriver-core.properties", ignoreResourceNotFound = true)
+class EurekaProviderConfiguration {
+  @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+  @Bean
+  @ConfigurationProperties("eureka.provider")
+  EurekaAccountConfigurationProperties eurekaConfigurationProperties() {
+    new EurekaAccountConfigurationProperties()
+  }
+
+  @Autowired
+  ObjectMapper objectMapper
+
+  @Autowired
+  EurekaApiFactory discoveryApiFactory
+
+  @Bean
+  EurekaCachingProvider eurekaCachingProvider(EurekaAccountConfigurationProperties eurekaAccountConfigurationProperties, List<EurekaAwareProvider> eurekaAwareProviderList) {
+    List<EurekaCachingAgent> agents = []
+    eurekaAccountConfigurationProperties.accounts.each { EurekaAccountConfigurationProperties.EurekaAccount accountConfig ->
+      accountConfig.regions.each { region ->
+        String discoveryHost = accountConfig.readOnlyUrl.replaceAll(Pattern.quote('{{region}}'), region)
```

Comment:
the old way was deriving regions from the account settings, but we're loading them independently here.